step-templates/iis-change-app-offline-online.json

```diff
@@ -3,10 +3,10 @@
   "Name": "IIS - Change App Offline",
   "Description": "Change the an app_offline file to app_online vice versa to turn the maintenance page on and off.",
   "ActionType": "Octopus.Script",
-  "Version": 7,
+  "Version": 8,
   "Properties": {
     "Octopus.Action.Script.Syntax": "PowerShell",
-    "Octopus.Action.Script.ScriptBody": "$offlineHtml = Join-Path $OctopusParameters[\"InstallationFolder\"] $OctopusParameters[\"AppOfflineFileName\"]\n$onlineHtml = Join-Path $OctopusParameters[\"InstallationFolder\"] $OctopusParameters[\"AppOnlineFileName\"]\n\nif (\"#{ChangeMode}\" -eq \"Online\")\n{\n    #Offline exists and so does online - remove offline\n    if ((Test-Path($offlineHtml)) -and (Test-Path($onlineHtml)))\n    {\n        Remove-Item $offlineHtml\n    }\n    \n    #Offline exists and online doesn't - move offline to online\n    if ((Test-Path($offlineHtml)) -and !(Test-Path($onlineHtml)))\n    {\n        Move-Item $offlineHtml $onlineHtml\n    }\n}\n\nif (\"#{ChangeMode}\" -eq \"Offline\")\n{\n    #Online exists and offline doesn't - move online to offline\n    if ((Test-Path($onlineHtml)) -and !(Test-Path($offlineHtml)))\n    {\n        Move-Item $onlineHtml $offlineHtml\n    }\n}"
+    "Octopus.Action.Script.ScriptBody": "$offlineHtml = Join-Path $OctopusParameters[\"InstallationFolder\"] $OctopusParameters[\"AppOfflineFileName\"]\n$onlineHtml = Join-Path $OctopusParameters[\"InstallationFolder\"] $OctopusParameters[\"AppOnlineFileName\"]\n\n#If neither file exists, throw a fit\nif ($OctopusParameters[\"CheckForFile\"] -and !(Test-Path($offlineHtml)) -and !(Test-Path($onlineHtml)))\n{\n\tWrite-Error \"Missing the online/offline files!\"\n    throw\n}\n\n\nif (\"#{ChangeMode}\" -eq \"Online\")\n{\n    #Offline exists and so does online - remove offline\n    if ((Test-Path($offlineHtml)) -and (Test-Path($onlineHtml)))\n    {\n        Remove-Item $offlineHtml\n    }\n    \n    #Offline exists and online doesn't - move offline to online\n    if ((Test-Path($offlineHtml)) -and !(Test-Path($onlineHtml)))\n    {\n        Move-Item $offlineHtml $onlineHtml\n    }\n}\n\nif (\"#{ChangeMode}\" -eq \"Offline\")\n{\n    #Online exists and offline doesn't - move online to offline\n    if ((Test-Path($onlineHtml)) -and !(Test-Path($offlineHtml)))\n    {\n        Move-Item $onlineHtml $offlineHtml\n    }\n}",
```

Comment:
The new value for `Octopus.Action.Script.ScriptBody` has a trailing comma `,` at the end of this line. This is causing a build failure.

Reply:
In the script addition:

```ps
#If neither file exists, throw a fit
if ($OctopusParameters["CheckForFile"] -and !(Test-Path($offlineHtml)) -and !(Test-Path($onlineHtml)))
{
	Write-Error "Missing the online/offline files!"
    throw
}
```

The `throw` isn't necessary, as the `Write-Error` will be written to the error stream and Octopus will detect this. You can swap the `throw` for a `return` statement.

In addition, the new parameter should be prefixed (see other comment) and should be tested for $True (more or readability) so the complete script should look something like this:

```ps
# If neither file exists, exit with error
if ($OctopusParameters["ChangeAppOffline.CheckForFile"] -eq $True -and !(Test-Path($offlineHtml)) -and !(Test-Path($onlineHtml)))
{
	Write-Error "Missing the online/offline files!"
  return
}
```